### ISSUE TITLE
Jenkins pods won't start (needs more free compute)

### ISSUE THREAD

**Issue by calum-cgi** [jenkins-x/jx#1462]
### Summary 
I have deployed jenkins-x to an Azure cluster using `jx create cluster aks`. Builds were running fine for a few days but recently stopped working. At the moment, pods are being created continuously and all are failing with the same event:

```
$ kubectl describe pod nodejs-t43vr

Events:
  Type     Reason            Age               From               Message
  ----     ------            ----              ----               -------
  Warning  FailedScheduling  54s (x3 over 2m)  default-scheduler  0/10 nodes are available: 10 Insufficient cpu.
```

Also from `jx status`:
```
Jenkins X installed for Cluster(#######): 10 nodes, memory 30% of 315801968Ki, cpu 95% of 79820m. Jenkins is running at #######.nip.io. Need more needs more free compute
```

I'm pretty confident that this isn't a CPU issue because I have upped the cluster to 10 x Standard D8s v3 (8 vcpus, 32 GB memory) which gives:

```
Total cluster capacity
Cores
80 vCPUs
Memory
320 GB
```

### Steps to reproduce the behavior
Start a build on any repository. Currently Python and Node.js repos are being built.

### Jx version

The output of `jx version` is:

```
Failed to get helm version: failed to run 'helm version --short' command in directory '', output: 'Error: unknown command "version --short" for "helm"
Run 'helm --help' for usage.': exit status 1
NAME               VERSION
jx                 1.3.79
jenkins x platform 0.0.1849
kubernetes cluster v1.9.9
kubectl            v1.11.0
git                git version 2.15.0.windows.1

? Would you like to upgrade to the new jx version? Yes
```

Seems to be an unrelated issue, but the upgrade fails with:
```
panic: gzip: invalid header
        panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x68 pc=0x5f36b9]
```

### Kubernetes cluster

What kind of Kubernetes cluster are you using & how did you create it?

Azure cluster created using `jx create cluster aks`. When pods stopped working, I increased the size of the cluster from 3 nodes to 10 nodes.

### Operating system / Environment
Windows 7 for `jx` command line tool and Azure cluster for Jenkins-x.
### Expected behavior
Pods should startup and run builds.
### Actual behavior
```Waiting for Pod to be scheduled (14/100): nodejs-j1588``` for every pod.

**Comment by jstrachan** [comment]
@calum-cgi  did you get any further with this? I wonder if its relating to the node size you picked ?

**Comment by calum-cgi** [comment]
I didn't get any further. My team was running out of time so I had to remove jenkins-x and just stick to standard jenkins with `helm install jenkins`. 

@ilyaa1 I'm using a paid for subscription for Azure. At first I thought the problem was lack of CPU so I tried with a massive cluster of 10 x Standard D8s v3 (8 vcpus, 32 GB memory) and had the same problem.

@jstrachan I tried with a few different node sizes. I don't know much about Kubernetes or how Jenkins-x is working but I tried the default of 3 nodes which were D3s. I also tried the D8s and D4s sizes.

When I ran `jx status` with different sized clusters, I always got the same result. Which was 95% CPU usage and the message `Need more needs more free compute`.

e.g.
`Jenkins X installed for Cluster(#######): 10 nodes, memory 30% of 315801968Ki, cpu 95% of 79820m. Jenkins is running at #######.nip.io. Need more needs more free compute`

My guess is that something went wrong installing Jenkins-x. I'm behind a corporate proxy server and tried running the `jx create cluster aks` command on Windows 7 and also in a Centos VM.

**Comment by thienlh** [comment]
I also having this issue after installed jenkins for several days. I was installed using `jx create cluster eks` which  create 2 m5 instaces on AWS, which is plenty so I don't think the problem is caused because of weak hardware.

**Comment by ccojocar** [comment]
This issue is related to the cluster size.

**Comment by calum-cgi** [comment]
@ccojocar what is the solution for this though? I've tried increasing my cluster size to 10 nodes and that didn't solve the problem. I had 80 vCPUs and 320 GB of memory.

**Comment by ccojocar** [comment]
@calum-cgi This is a quite large cluster.  Wondering what is your node size? Can you try with larger nodes? How many builds do you run per day?

From your description, it looks like the Kubernetes scheduler is not able to allocate enough CPU for Jenkins X PODs. 

You can also run `kubectl describe pod <pod-name>' , to check where it is stuck.

**Comment by calum-cgi** [comment]
@ccojocar I tried up to `Standard D8s v3` which have 8 CPUs and 32GB of memory each. Each time this happened, I rebuilt the cluster with larger nodes up to the D8s v3 which were very expensive. We ran around 5-10 builds per day - our project had just started up and we only had a couple of repos.

This was a while ago now so we eventually just gave up and installed [jenkins with Helm](https://github.com/helm/charts/tree/master/stable/jenkins) which works for us really well with just 3 x Standard D4s v3 (4 vcpus, 16 GB memory) totalling 12 vCPUs and 48GB of memory.

`kubectl describe pod` at the time would output the same thing no matter what sized cluster we used:
```
Events:
  Type     Reason            Age               From               Message
  ----     ------            ----              ----               -------
  Warning  FailedScheduling  54s (x3 over 2m)  default-scheduler  0/10 nodes are available: 10 Insufficient cpu.
```

It was a very strange problem. Each time it happened, I would have to completely re-deploy jenkins-x onto a larger cluster. A few days later all the builds would stop and I would see the same errors. Since deploying standard jenkins with helm, we have had no issues over the past 4 months.

Thanks for looking into this issue. I am keen to use jenkins-x on my next project if this issue is resolved.
